trieste/acquisition/function.py

```diff
@@ -1116,64 +1299,96 @@ def prepare_acquisition_function(
             or ``dataset`` is empty.
         """
         tf.debugging.assert_positive(len(dataset))
+        tf.debugging.Assert(None in [pending_points], [])
 
-        if (
-            pending_points is None
-        ):  # compute penalization params and base acquisition once per optimization step
-            samples = self._search_space.sample(num_samples=self._num_samples)
-            samples = tf.concat([dataset.query_points, samples], 0)
-
-            def get_lipschitz_estimate(
-                sampled_points: TensorType,
-            ) -> tf.Tensor:  # use max norm of posterior mean gradients
-                with tf.GradientTape() as g:
-                    g.watch(sampled_points)
-                    mean, _ = model.predict(sampled_points)
-                grads = g.gradient(mean, sampled_points)
-                grads_norm = tf.norm(grads, axis=1)
-                max_grads_norm = tf.reduce_max(grads_norm)
-                eta = tf.reduce_min(mean, axis=0)
-                return max_grads_norm, eta
-
-            lipschitz_constant, eta = get_lipschitz_estimate(samples)
-            if (
-                lipschitz_constant < 1e-5
-            ):  # threshold to improve numerical stability for 'flat' models
-                lipschitz_constant = 10
-
-            self._lipschitz_constant = lipschitz_constant
-            self._eta = eta
-
-            if isinstance(self._base_builder, ExpectedImprovement):  # reuse eta estimate
-                self._base_acquisition_function = expected_improvement(model, self._eta)
-            else:
-                self._base_acquisition_function = self._base_builder.prepare_acquisition_function(
-                    dataset, model
-                )
+        # no penalization required if no pending_points.
+        return self._update_base_acquisition_function(dataset, model)
 
-        tf.debugging.Assert(
-            None not in (self._lipschitz_constant, self._base_acquisition_function), []
-        )
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        dataset: Dataset,
+        model: ProbabilisticModel,
+        pending_points: Optional[TensorType] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param dataset: The data from the observer.
+        :param model: The model over the specified ``dataset``.
+        :param pending_points: Points already chosen to be in the current batch (of shape [M,D]),
+            where M is the number of pending points and D is the search space dimension.
+        :return: The updated acquisition function.
+        """
+        tf.debugging.assert_positive(len(dataset))
+        tf.debugging.Assert(None not in [self._base_acquisition_function], [])
```

Comment:
What does this line do?

Reply:
Checks that the base acquisition function already been constructed by a previous call to prepare_acquisition_function. The funny syntax is to make tensorflow happy (and is copied from similar checks elsewhere in this file).